docs/writing_plugins.rst

```diff
@@ -17,67 +20,73 @@ There are four types of helper functions:
 
 Each helper function type modifies a different `apispec.APISpec` method. For example, path helpers modify `apispec.APISpec.add_path`.
 
-A helper function may look something like this:
+
+A plugin with a path helper function may look something like this:
 
 .. code-block:: python
 
-    from apispec import Path
+    from apispec import Path, BasePlugin
     from apispec.utils import load_operations_from_docstring
 
-    def docstring_path_helper(spec, path, func, **kwargs):
-        """Path helper that parses docstrings for operations. Adds a
-        ``func`` parameter to `apispec.APISpec.add_path`.
-        """
-        operations = load_operations_from_docstring(func.__doc__)
-        return Path(path=path, operations=operations)
+    class MyPlugin(BasePlugin):
 
+        def path_helper(self, path, func, **kwargs):
+            """Path helper that parses docstrings for operations. Adds a
+            ``func`` parameter to `apispec.APISpec.add_path`.
+            """
+            operations = load_operations_from_docstring(func.__doc__)
+            return Path(path=path, operations=operations)
 
-The ``setup`` Function
-----------------------
 
-All plugins **must** define a ``setup`` function that receives an `apispec.APISpec` instance as its only argument. The ``setup`` function registers your plugin's helper functions with the ``spec``.
+The ``init_spec`` Method
+------------------------
 
+`BasePlugin` has an `init_spec` method that `APISpec` calls on each plugin at initialization with the spec object itself as parameter. It is a no-op, but it can be overriden by a plugin to get useful information from the spec object.
```

Comment:
Revision suggestion: "It is no-op by default, but a plugin may override it to access and store useful information from the spec object."